app/src/main/java/fr/free/nrw/commons/explore/depictions/SearchDepictionsFragment.java

```diff
@@ -0,0 +1,238 @@
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
+public class SearchDepictionsFragment extends CommonsDaggerSupportFragment {
+    private static int TIMEOUT_SECONDS = 15;
+
+    @BindView(R.id.imagesListBox)
+    RecyclerView depictionsRecyclerView;
+    @BindView(R.id.imageSearchInProgress)
+    ProgressBar progressBar;
+    @BindView(R.id.imagesNotFound)
+    TextView depictionNotFound;
+    String query;
+    @BindView(R.id.bottomProgressBar)
+    ProgressBar bottomProgressBar;
+    boolean isLoadingDepictions;
+    @Inject
+    RecentSearchesDao recentSearchesDao;
+    @Inject
+    MediaWikiApi mwApi;
+    @Inject
+    DepictsClient depictsClient;
+    @Inject
+    @Named("default_preferences")
+    JsonKvStore basicKvStore;
+    private RVRendererAdapter<DepictedItem> depictionsAdapter;
+    private List<DepictedItem> queryList = new ArrayList<>();
+
+    private final SearchDepictionsAdapterFactory adapterFactory = new SearchDepictionsAdapterFactory(item -> {
+        // Called on Click of a individual depicted Item
+        // Open Depiction Details activity
+        DepictedImagesActivity.startYourself(getContext(), item);
+        saveQuery(query);
+    });
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_browse_image, container, false);
+        ButterKnife.bind(this, rootView);
+        if (getActivity().getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT) {
+            depictionsRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+        } else {
+            depictionsRecyclerView.setLayoutManager(new GridLayoutManager(getContext(), 2));
+        }
+        ArrayList<DepictedItem> items = new ArrayList<>();
+        depictionsAdapter = adapterFactory.create(items);
+        depictionsRecyclerView.setAdapter(depictionsAdapter);
+        depictionsRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
+                // check if end of recycler view is reached, if yes then add more results to existing results
+                if (!recyclerView.canScrollVertically(1)) {
+                    addDepictionsToList(query);
+                }
+            }
+        });
+        return rootView;
+    }
+
+    /**
+     * Adds 25 more results to existing search results
+     */
+
+    private void addDepictionsToList(String query) {
+        if (isLoadingDepictions) return;
+        isLoadingDepictions = true;
+        this.query = query;
+        bottomProgressBar.setVisibility(View.VISIBLE);
+        progressBar.setVisibility(GONE);
+        compositeDisposable.add(depictsClient.searchForDepictions(query, 25, queryList.size())
```

Comment:
Where have you cleared the composite disposable.

Reply:
is present in the recent commit